plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/logging/RemoteQueryModifierConfig.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc.logging;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.trino.plugin.jdbc.logging.RemoteQueryModifier.PredefinedValue;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.MatchResult;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.stream.Collectors.joining;
+
+public class RemoteQueryModifierConfig
+{
+    private static final List<String> PREDEFINED_VALUE_CODES = Arrays.stream(PredefinedValue.values()).map(PredefinedValue::getPredefinedValueCode).toList();
+    private static final List<String> PREDEFINED_MATCHES = Arrays.stream(PredefinedValue.values()).map(PredefinedValue::getMatchCase).toList();
+    private static final Pattern VALIDATION_PATTERN = Pattern.compile("[\\w ,=]|" + String.join("|", PREDEFINED_MATCHES));
+    private String format = "";
+
+    @Config("remote-query.comment.format")
+    @ConfigDescription("Format in which logs about query execution context should be added as comments sent through jdbc driver." +
+            " It can consist only of letters, digits, spaces, underscores, equality signs and predefined values: $QUERY_ID, $SOURCE, $USER, $TRACE_TOKEN")
+    public RemoteQueryModifierConfig setFormat(String format)
+    {
+        this.format = format;
+        return this;
+    }
+
+    public String getFormat()
+    {
+        return format;
+    }
+
+    void validate()
```

Comment:
I've had issues with @Pattern annotation, as RegExp in java are stoping after a first match in the source and are not trying to repeat the match in the rest of the source. This makes writing a single RegExp very difficult (at least for me), that would allow for checking repeated patterns. In Java this case of matching repeatedly the pattern against whole source is solved by the methods, not by the RegExp flags itself.